### PR TITLE
[BE][Endpoint] Add New Endpoint for Listing Leaderboard By Game Id

### PR DESCRIPTION
**Pull request by grgsferry**
## Description
**DOD:**

- to create new endpoint (`GET: /api/v1/games/leaderboard/:gameId/`)
- with no authentication
- with response
  - ```json
    ‌{
        "success": true,
        "message": "Success Get Games Leaderboard",
        "data": [
          {
            "name": "Player 1",
            "email": "email@email.com",
            "badge": "Gold",
            "points": 1000
          },
          {
            "name": "Player 12",
            "email": "email@email.com",
            "badge": "Gold",
            "points": 999
          },
          {
            "name": "Player 24",
            "email": "email@email.com",
            "badge": "Gold",
            "points": 998
          }
        ],
        "error": null
    }
    ```

**Hint:**
- final SQL query:

```sql
SELECT CONCAT(b.first_name, ' ', b.last_name) AS name, 
b.email, CASE WHEN c.badge_name IS NOT NULL THEN c.badge_name ELSE 'Bronze' END AS badge, SUM(a.points_earned) AS points 
FROM game_histories a 
LEFT JOIN users b ON a.player_id = b.id 
LEFT JOIN user_badges_history c ON a.player_id = c.user_id 
WHERE a.game_id = ${id} 
GROUP BY 1,2,3 
ORDER BY 4 DESC
```
‌
- Trello Link: [here](https://trello.com/c/XgcG2XzN/20-beendpoint-add-new-endpoint-for-listing-leaderboard-by-game-id)

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Refactor/Breaking Change

### CODE REVIEW

File: src/domains/games/router.js [hunk 1]
```diff
@@ -16,6 +16,7 @@ const routes = {
     validation(gameValidator.paramsId),
     gameController.addPlayCount,
   ],
+  'GET: /leaderboard/:id': [validation(gameValidator.paramsId), gameController.getLeaderboard],
```
Comment (Nursalli) [inline src/domains/games/router.js:20]: validator buat params id nya ngga ada mas?

File: src/domains/games/repository.js [hunk 2]
```diff
@@ -21,4 +21,11 @@ module.exports = {
       [param]: updatedGame[param],
     };
   },
+  getLeaderboard: async (id) => {
+    const result = await sequelize.query(
+      `SELECT CONCAT(b.first_name, ' ', b.last_name) AS name, b.email, CASE WHEN c.badge_name IS NOT NULL THEN c.badge_name ELSE 'Bronze' END AS badge, SUM(a.points_earned) AS points FROM game_histories a LEFT JOIN users b ON a.player_id = b.id LEFT JOIN user_badges_history c ON a.player_id = c.user_id WHERE a.game_id = ${id} GROUP BY 1,2,3 ORDER BY 4 DESC`
+    );
```
Comment (Nursalli) [inline src/domains/games/repository.js:33]: mungkin tambahin kondisi jika ngga exist game id ny mas

File: src/domains/games/router.js (hunk 1 above)
Comment (grgsferry) [reply src/domains/games/router.js:20]: validator paramsId nya aku make yang udah masuk ke commit sebelumnya mass @Nursalli

File: src/domains/games/repository.js (hunk 2 above)
Comment (grgsferry) [reply src/domains/games/repository.js:33]: okeokee coba kuexplore dulu gimana kalo make raw query ya mas.
thankyou

File: src/domains/games/router.js (hunk 1 above)
Comment (Nursalli) [reply src/domains/games/router.js:20]: Sipp

File: src/domains/games/repository.js (hunk 2 above)
Comment (hrz8) [reply src/domains/games/repository.js:33]: mungkin bisa

```js
if (!result || !result[0]) {
  throw new AppError('Game doesnt exists', 404);
}
```

cc @grgsferry

Comment (grgsferry) [reply src/domains/games/repository.js:33]: thankyou mas hirzi @hrz8

**Review by hrz8 — APPROVED** [review]

**Review by irvanariyanto — APPROVED** [review]

**Pull request merged**